dcicutils/glacier_utils.py

```diff
@@ -0,0 +1,455 @@
+                   If you want to use the internal methods, see:
+                       * restore_portal_from_glacier
+                       * copy_object_back_to_original_location
+                       * patch_file_lifecycle_status_to_standard
+                       * delete_glaciered_object_versions
+        """
+        self.s3 = boto3.client('s3')
+        self.env_name = env_name
+        self.key_manager = CGAPKeyManager()
+        self.env_key = self.key_manager.get_keydict_for_env(env_name)
+        self.health_page = get_health_page(key=self.env_key, ff_env=env_name)
+
+    def resolve_possible_file_status(self) -> list:
+        """ Checks the File.json profile to see valid status values for files """
+        profile = get_metadata('/profiles/file.json', key=self.env_key, ff_env=self.env_name)
+        return profile['properties']['status']['enum']
+
+    def resolve_possible_lifecycle_status(self) -> list:
+        profile = get_metadata('/profiles/file.json', key=self.env_key, ff_env=self.env_name)
+        return profile['properties']['s3_lifecycle_status']['enum']
+
+    def resolve_possible_lifecycle_category(self) -> list:
+        profile = get_metadata('/profiles/file.json', key=self.env_key, ff_env=self.env_name)
+        return profile['properties']['s3_lifecycle_category']['suggested_enum']
+
+    def resolve_bucket_key_from_portal(self, atid: str, atid_meta: dict = None) -> List[Tuple[str, str]]:
+        """ Resolves the bucket, key combination for the given @id
+            Raises GlacierRestoreException if not found
+
+        :param atid: resource path to extract bucket, key information from
+        :param atid_meta: metadata if already resolved (such as from search)
+        :return: bucket, key tuple
+        """
+        if not atid_meta:
+            atid_meta = get_metadata(atid, key=self.env_key, ff_env=self.env_name,
+                                     add_on='frame=object&datastore=database')
+        upload_key = atid_meta.get('upload_key', None)
+        atid_types = atid_meta.get('@type', [])
+        bucket = self.health_page.get('file_upload_bucket')
+        if upload_key:
+            if 'FileProcessed' in atid_types:
+                bucket = self.health_page.get('processed_file_bucket')
+                files = [(bucket, upload_key)]
+            else:  # if not a processed file assume it is an uploaded file
+                files = [(bucket, upload_key)]
+        else:
+            raise GlacierRestoreException(f'@id {atid} does not have an upload_key, thus cannot be queried for'
+                                          f' Glacier restore.')
+        # Add extra files
+        if 'extra_files' in atid_meta:
+            for extra_file in atid_meta['extra_files']:
+                if 'upload_key' in extra_file:
+                    files.append((bucket, extra_file['upload_key']))
+        return files
+
+    def restore_portal_from_glacier(self, atid: str, file_meta: Union[None, dict] = None,
```

Comment:
Consider renaming this method for clarity, perhaps something like `get_portal_file_and_restore_from_glacier`.